### ISSUE TITLE
[1.5] network.inner, observer.ingress and observer.egress object definitions

### ISSUE THREAD

**Issue by codebrain** [elastic/ecs#780]
The following new schema definitions in ECS 1.5:

- `network.inner`
- `observer.ingress`
- `observer.egress`

Exhibit the same format of definition in the `ecs_nested.yml` file as:

- `dns.anwers`
- `log.syslog`

That is, they specify the root object and then subsequent properties off of this root object, for example (truncated YAML sample):

```
ingress:
      dashed_name: observer-ingress
      description: Observer.ingress holds information like interface number and name,
        vlan, and zone information to  classify ingress traffic.  Single armed monitoring
        such as a network sensor on a span port should  only use observer.ingress
        to categorize traffic.
      type: object
ingress.interface.alias:
      dashed_name: observer-ingress-interface-alias
      description: Interface alias as reported by the system, typically used in firewall
        implementations for e.g. inside, outside, or dmz logical interface naming.
      type: keyword
ingress.interface.id:
      dashed_name: observer-ingress-interface-id
      description: Interface ID as reported by an observer (typically SNMP interface
        ID).
      type: keyword
```

In C# it is not possible to have both a `class` and `property` with the same name, so in the ECS .NET generator we have adopted the same strategy for these new properties as `dns.answers` and `log.syslog`.

Does this seem reasonable?

**Comment by webmat** [comment]
Can you remind me what that strategy is? :-)

**Comment by codebrain** [comment]
Sure, for example the approach we take with `observer.ingress` is:

1. Ignore the `ingress` YAML property marked as `object` because it has child properties of `ingress.interface.*`, `ingress.vlan.*` and `zone`.
2. Generate C# classes for the child hierarchy, to include `ObserverIngress`, `ObserverVlan` as shown:
https://github.com/elastic/ecs-dotnet/blob/280dd17bc95a968013f2d65bc3592d7742e8173b/src/Elastic.CommonSchema/Types.Generated.cs#L2614-L2813
3. Add a property to `Observer` named `Ingress` of type `ObserverIngress`.
4. Copy contents of description from `ingress` YAML to property of `Observer.Ingress`.

This way we can keep the description given in the YAML file for `ingress` and then apply it to the `Ingress` property on the `Observer` class.

**Comment by codebrain** [comment]
Also, is `log.syslog` missing the `normalize->array` definition? Should it be an array?

**Comment by webmat** [comment]
Yes I think this approach makes sense.

We defined these fields as datatype `object` only to have a target for the nesting of `vlan` and `interface` field sets. We did this because it was perfectly equivalent in Elasticsearch -- it generates the exact same template whether we do this or not.

So you skipping the Object definitions whenever there's subfields defined also works well, I think. Make sure you add the Object definitions when there's no other fields defined under it, however. I'm thinking of `labels` here; I think that's the only `object` field we have that has nothing defined under it.
